src/transport/tcp/mod.rs

```diff
@@ -201,6 +202,7 @@ impl TcpTransport {
 
         match dial_addresses.local_dial_address(&remote_address.ip()) {
             Ok(Some(dial_address)) => {
+                socket.set_nodelay(nodelay)?;
```

Comment:
Why we set `TCP_NODELAY` on a socket only if we bind to specific addr/port? Shouldn't it be set in all cases?

Reply:
Oversight on my part, good catch. Fixed